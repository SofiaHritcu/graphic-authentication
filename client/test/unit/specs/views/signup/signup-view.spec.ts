import { createLocalVue, shallowMount } from "@vue/test-utils";
import SignUpView from "@/views/signup/signup-view.vue";
import "regenerator-runtime/runtime";

describe("SignUp View", () => {
  let wrapper: any;

  beforeEach(() => {
    // {
    //   mocks: {
    //     $router: {
    //       push: jest.fn(),
    //     },
    //     $vuetify: {
    //       breakpoint: {
    //         name: "",
    //       },
    //     },
    //   },
    // }
    wrapper = shallowMount(SignUpView, {
      mocks: {
        $router: {
          push: jest.fn(),
        },
        $vuetify: {
          breakpoint: {
            name: "",
          },
        },
      },
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  describe("snapshot", () => {
    it("should match snapshot", () => {
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });

  describe("methods", () => {
    describe("handleSignupCardClick", () => {
      it("should execute router push with the right parameters", async () => {
        expect.assertions(1);
        const spyRouterPush = jest.spyOn(wrapper.vm.$router, "push");

        await wrapper.vm.handleSignupCardClick("ga-signup");

        expect(spyRouterPush).toHaveBeenCalledWith({ name: "ga-signup" });
      });
    });
    describe("handleAuthCardClick", () => {
      it("should execute router push with the right parameters", async () => {
        expect.assertions(1);
        const spyRouterPush = jest.spyOn(wrapper.vm.$router, "push");

        await wrapper.vm.handleAuthCardClick("ga-signup");

        expect(spyRouterPush).toHaveBeenCalledWith({ name: "ga-signup" });
      });
    });
    describe("handleIPassClick", () => {
      it("should execute router push with welcome route name", async () => {
        expect.assertions(1);
        const spyRouterPush = jest.spyOn(wrapper.vm.$router, "push");

        await wrapper.vm.handleIPassClick();

        expect(spyRouterPush).toHaveBeenCalledWith({ name: "ga-welcome" });
      });
    });
  });

  describe("computed", () => {
    describe("avatarSize", () => {
      it("should return 100 when being on smaller viewports", () => {
        wrapper.vm.$vuetify.breakpoint.name = "xs";
        expect(wrapper.vm.avatarSize).toEqual(100);
      });
      it("should return 150 when being on larger viewports", () => {
        wrapper.vm.$vuetify.breakpoint.name = "lg";
        expect(wrapper.vm.avatarSize).toEqual(150);
      });
    });
  });
});
