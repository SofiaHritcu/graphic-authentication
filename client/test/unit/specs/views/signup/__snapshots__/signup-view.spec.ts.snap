// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp View snapshot should match snapshot 1`] = `
<div
  class="ga__signup_view"
>
  <app-menu-stub
    color="teal"
  />
   
  <v-row
    align-content="center"
    class="ga__signup_view__title"
    justify="center"
  >
    <v-col
      cols="12"
    >
      <div
        class="text-md-h5 grey--text text--darken-1"
      >
        Choose an
      </div>
       
      <div
        class="text-overline font-weight-light white--text ga__signup_view__title__text"
      >
        
          IPass
        
      </div>
    </v-col>
  </v-row>
   
  <v-row
    align-content="center"
    class="ga__signup_view__auth_cards"
  >
    <v-col
      class="ga__signup_view__auth_cards__container"
      cols="6"
      md="3"
      sm="3"
      xs="6"
    >
      <v-card
        class="ga__signup_view__auth_cards__card ga__signup_view__auth_cards__card--teal"
      >
        <v-avatar
          class="ga__signup_view__auth_cards__card__avatar"
          color="white"
          size="150"
        >
          <img
            src=""
          />
        </v-avatar>
         
        <v-card-text
          class="text-overline ga__signup_view__auth_cards__card__title"
        >
          
            Icon Pass
          
        </v-card-text>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="teal darken-2"
          outlined=""
        >
          
            signup
          
        </v-btn>
         
        <div
          class="text-overline blue-grey--text mt-2"
        >
          Already have an account?
        </div>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="teal darken-2"
          small=""
          text=""
        >
          
            login
          
        </v-btn>
      </v-card>
    </v-col>
    <v-col
      class="ga__signup_view__auth_cards__container"
      cols="6"
      md="3"
      sm="3"
      xs="6"
    >
      <v-card
        class="ga__signup_view__auth_cards__card ga__signup_view__auth_cards__card--indigo"
      >
        <v-avatar
          class="ga__signup_view__auth_cards__card__avatar"
          color="white"
          size="150"
        >
          <img
            src=""
          />
        </v-avatar>
         
        <v-card-text
          class="text-overline ga__signup_view__auth_cards__card__title"
        >
          
            Other Pass
          
        </v-card-text>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="indigo darken-2"
          outlined=""
        >
          
            signup
          
        </v-btn>
         
        <div
          class="text-overline blue-grey--text mt-2"
        >
          Already have an account?
        </div>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="indigo darken-2"
          small=""
          text=""
        >
          
            login
          
        </v-btn>
      </v-card>
    </v-col>
    <v-col
      class="ga__signup_view__auth_cards__container"
      cols="6"
      md="3"
      sm="3"
      xs="6"
    >
      <v-card
        class="ga__signup_view__auth_cards__card ga__signup_view__auth_cards__card--deep-orange"
      >
        <v-avatar
          class="ga__signup_view__auth_cards__card__avatar"
          color="white"
          size="150"
        >
          <img
            src=""
          />
        </v-avatar>
         
        <v-card-text
          class="text-overline ga__signup_view__auth_cards__card__title"
        >
          
            Other Pass
          
        </v-card-text>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="deep-orange darken-2"
          outlined=""
        >
          
            signup
          
        </v-btn>
         
        <div
          class="text-overline blue-grey--text mt-2"
        >
          Already have an account?
        </div>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="deep-orange darken-2"
          small=""
          text=""
        >
          
            login
          
        </v-btn>
      </v-card>
    </v-col>
    <v-col
      class="ga__signup_view__auth_cards__container"
      cols="6"
      md="3"
      sm="3"
      xs="6"
    >
      <v-card
        class="ga__signup_view__auth_cards__card ga__signup_view__auth_cards__card--yellow"
      >
        <v-avatar
          class="ga__signup_view__auth_cards__card__avatar"
          color="white"
          size="150"
        >
          <img
            src=""
          />
        </v-avatar>
         
        <v-card-text
          class="text-overline ga__signup_view__auth_cards__card__title"
        >
          
            Other Pass
          
        </v-card-text>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="yellow darken-2"
          outlined=""
        >
          
            signup
          
        </v-btn>
         
        <div
          class="text-overline blue-grey--text mt-2"
        >
          Already have an account?
        </div>
         
        <v-btn
          class="ga__signup_view__auth_cards__card__button"
          color="yellow darken-2"
          small=""
          text=""
        >
          
            login
          
        </v-btn>
      </v-card>
    </v-col>
  </v-row>
</div>
`;
